test/util/unit_test.cpp

```diff
@@ -413,29 +413,22 @@ class WildcardFilter : public Filter {
 class IntraTestLogger : public Logger {
 public:
     // Logger with a local log level used for the different test threads
+    // A new log level threshold atomic will be used for this logger, but
+    // it will write to the base_logger_ptr when do_log() is called
     IntraTestLogger(const std::shared_ptr<Logger>& base_logger_ptr, Level threshold)
```

Comment:
can you write a real quick little test that instantiates an IntraTestLogger at like trace or debug level and then verify that it actually respects that?

Reply:
Added test, but I needed to move the `IntraTestLogger` to _logger.hpp_ - I renamed the class to `LocalThresholdLogger`